etl/steps/data/garden/antibiotics/2024-11-12/antimicrobial_usage.py

```diff
@@ -31,14 +38,128 @@ def run(dest_dir: str) -> None:
 
     tb_class = tb_class.format(["country", "year", "antimicrobialclass", "atc4name", "routeofadministration"])
     tb_aware = tb_aware.format(["country", "year", "awarelabel"])
+    tb_class_agg = pivot_aggregated_table(tb_class_agg, tb_notes)
+    tb_class_agg = tb_class_agg.format(["country", "year"], short_name="class_aggregated")
 
     #
     # Save outputs.
     #
     # Create a new garden dataset with the same metadata as the meadow dataset.
     ds_garden = create_dataset(
-        dest_dir, tables=[tb_class, tb_aware], check_variables_metadata=True, default_metadata=ds_meadow.metadata
+        dest_dir,
+        tables=[tb_class, tb_aware, tb_class_agg],
+        check_variables_metadata=True,
+        default_metadata=ds_meadow.metadata,
     )
 
     # Save changes in the new garden dataset.
     ds_garden.save()
+
+
+def pivot_aggregated_table(tb_class_agg: Table, tb_notes: Table) -> Table:
+    """
+    Pivot the aggregated table to have a column for each antimicrobial class, then add the description_processing metadata
+    """
+
+    tb_notes_dict = {
+        "Antibacterials (ATC J01, A07AA, P01AB)": "antibacterials",
+        "Antimalarials (ATC P01B)": "anti_malarials",
+        "Antimycotics and antifungals for systemic use (J02, D01B)": "antifungals",
+        "Antivirals for systemic use (ATC J05)": "antivirals",
+        "Drugs for the treatment of tuberculosis (ATC J04A)": "antituberculosis",
+        "Antibacterials (ATC J01, A07AA, P01AB, ATC J04A)": "antibacterials_and_antituberculosis",
+    }
+    tb_notes["category"] = tb_notes["antimicrobialclass"].map(tb_notes_dict)
+    tb_class_agg = tb_class_agg.copy(deep=True)
+    tb_class_agg["antimicrobialclass"] = tb_class_agg["antimicrobialclass"].replace(tb_notes_dict)
+    tb_class_agg = tb_class_agg.pivot(index=["country", "year"], columns="antimicrobialclass", values=["ddd", "did"])
+    tb_class_agg.columns = tb_class_agg.columns.to_flat_index()
+    tb_class_agg.columns = [f"{col[0]}_{col[1]}" for col in tb_class_agg.columns]
+    tb_class_agg = tb_class_agg.reset_index()
+
+    for key in tb_notes_dict.values():
+        if f"ddd_{key}" in tb_class_agg.columns:
+            tb_class_agg[f"ddd_{key}"].metadata.description_processing = tb_notes["description_processing"][
+                tb_notes["category"] == key
+            ]
+        if f"did_{key}" in tb_class_agg.columns:
+            tb_class_agg[f"did_{key}"].metadata.description_processing = tb_notes["description_processing"][
+                tb_notes["category"] == key
+            ].astype(str)
```

Comment:
I see that description_processing is [not being rendered well](http://staging-site-amu-agg/etl/wizard?page=chart-diff&chart_id=8323) in grapher.
This probably happens because `description_processing` should be a string (which can contain bullet points, as you do), not a list.
![Screenshot 2024-11-28 at 12 59 45](https://github.com/user-attachments/assets/2bed35c2-d76b-4da0-8ccd-c610a26def70)

Reply:
I see that you attempted to fix this on `format_notes`, but in the end when you create the metadata field here, you'd still need to convert the list of items into a string, with something like:
```
"\n".join(tb_notes["description_processing"][
                tb_notes["category"] == key
            ].tolist())
```
(I haven't tried, but I guess you see my point)